### PR TITLE
refactor: remove recursive methods

### PR DESCRIPTION
**Pull request by Beraliv**
Connected to [Issue#136](https://github.com/nanoutils/nanoutils/issues/136)
:fire: Ready for merge

### CODE REVIEW

File: lib/unnestN/index.js [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
-      })
-      ? arr
-      : unnestN(num - 1, unnest(arr))
-    : []
+  if (!Array.isArray(arr) || num < 0) {
+    return []
+  }
+  var res = arr
+  var i = 0
+  while (i < num && !res.every(compose(not, Array.isArray))) {
```
Comment (BabkinAleksandr) [inline lib/unnestN/index.js]: I think it's much more expensive to check `!res.every(compose(not, Array.isArray))`, than just let it do several more `concat` operations in `unnest` method
It's worth checking

File: lib/flattenObj/index.js [hunk 2]
```diff
@@ -2,15 +2,15 @@ import a2p from '../_internal/_a2p'
...
+export default nAry(1, function flattenObj(obj, nest) {
   var nestA = p2a(nest)
   return Object.keys(obj).reduce(function(res, key) {
     if (
       obj[key] !== null &&
       typeof obj[key] === 'object' &&
       !Array.isArray(obj[key])
     ) {
-      var n = objToFlat(obj[key], nestA.concat(key))
+      var n = flattenObj(obj[key], nestA.concat(key))
```
[the remark this answers is not in the document]
Comment (Beraliv) [reply lib/flattenObj/index.js]: It needs to change it completely so I postponed it to a weekend

[the remark this answers is not in the document]
Comment (BabkinAleksandr) [reply lib/flattenObj/index.js]: ok, good 👍

File: lib/unnestN/index.js (hunk 1 above)
Comment (Beraliv) [reply lib/unnestN/index.js]: also I'm thinking of moving `compose(not, Array.isArray)` away from `every` as it's created over and over again for each iteration (maybe)
at least I need to check it twice

Comment (BabkinAleksandr) [reply lib/unnestN/index.js]: Yep, that's the only truthy way

Comment (BabkinAleksandr) [reply lib/unnestN/index.js]: > also I'm thinking of moving `compose(not, Array.isArray)` away from `every` as it's created over and over again for each iteration (maybe)
> at least I need to check it twice

Good thought

File: lib/repeat/index.js [hunk 3]
```diff
@@ -1,5 +1,13 @@
...
 
 export default curry2(function repeat(val, idx) {
-  return idx === 0 ? [] : [val].concat(repeat(val, idx - 1))
+  if (idx < 1) {
+    return []
+  }
+  var arr = Array(idx)
+  var i = 0
+  while (i < idx) {
+    arr[i++] = val
```
[the remark this answers is not in the document]
Comment (Beraliv) [reply lib/repeat/index.js]: The different is not visible
screen shot 2018-09-14 at 23 45 02

**Pull request merged**